src/OrbitBase/ThreadUtilsLinux.cpp

```diff
@@ -40,12 +40,15 @@ std::string GetThreadName(pid_t tid) {
   return comm_content.value();
 }
 
-void SetCurrentThreadName(const std::string& thread_name) {
-  std::string name = thread_name;
-  if (name.length() >= kMaxThreadNameLength) {
-    name[kMaxThreadNameLength - 1] = '\0';
+void SetCurrentThreadName(std::string_view thread_name) {
+  std::string buf;
+  const char* thread_name_cstr = thread_name.data();
+  if (thread_name.length() >= kMaxThreadNameLength) {
+    buf = thread_name.substr(0, kMaxThreadNameLength - 1);
+    thread_name_cstr = buf.data();
```

Comment:
same thing here.

Reply:
well, this part is fine because std::strings c_str() are null-terminated.

Reply:
well c_str() yes, but not data() I think, no?

Reply:
actually it seems data does return a null terminated string https://en.cppreference.com/w/cpp/string/basic_string/data#:~:text=The%20returned%20array%20is%20null,to%20a%20single%20null%20character.&text=The%20pointer%20obtained%20from%20data,any%20standard%20library%20function%2C%20or